cou/cli.py

```diff
@@ -131,21 +125,19 @@ async def analyze_and_plan(
     progress_indicator.succeed()
 
     progress_indicator.start("Generating upgrade plan...")
-    upgrade_plan = await generate_plan(analysis_result, backup_database)
+    upgrade_plan = await generate_plan(analysis_result, args)
```

Comment:
One of disadvantage of this is, that I do not know what arguments are really needed by function and that we are passing more that it's needed (e.g. `args.verbosity`, `args.command`, ...)

Reply:
Yes, but at the same time I prefer to know that the function is using something that comes from the CLI arguments instead of `backup_database`. For someone that never saw the code, looking just at this argument it's almost impossible to figure out that came from the user's input

Reply:
I'll not say impossible, but anyway mixing logic like this in most case is more harmful than useful.

Reply:
I agree with Robert on this. When going through the changes, I find it strange that the entirety of `CLIargs` is passed to `generate_plan`, in which only one value is actually needed. I think `backup_database` is clearer, and people can simply go back to the calling function to see where the value come from.

This is also true for [`get_log_level`](https://github.com/canonical/charmed-openstack-upgrader/blob/main/cou/cli.py#L75) and [`COUModel()`](https://github.com/canonical/charmed-openstack-upgrader/blob/main/cou/cli.py#L122), which I think it's good that you kept unchanged.

Reply:
Since this will never be used as Python package it's fine, but if I'm wrong and some will use it as Python package how that will look like? IMO it will be really strange to use, for example generating plan, and you need to provide full `CLIargs` object, where there will be parts which are not used at all.


I do not want to block this, so I'll resolve this conversation.

Reply:
@agileshaw with the next changes we would need to pass more than just `backup_database`. [ pre_plan_sanity_checks](https://github.com/canonical/charmed-openstack-upgrader/pull/222/files#diff-abbb76655f6d778c97d8958cb0158d4fcc2f87854044f43695bcc788b76489e5R211) will need to have `upgrade_groups` and also `machines`, `hosrnames`, `avaiabilty-zones` and etc. The number of inputs to generate plan will start to be quiet big and against the clean code principles.

@rgildein as far as I know, just the map between workload and OpenStack release might be a python package in the future.

Reply:
I'm sorry, but what is "against the clean code principles"? I'm not aware of any official guideline like this in any PEP.